src/daft-functions/src/binary/concat.rs

```diff
@@ -0,0 +1,129 @@
+                        format_field_type_for_error(&right_field),
+                    ))),
+                }
+            }
+            _ => Err(DaftError::SchemaMismatch(format!(
+                "Expected 2 input args, got {}",
+                inputs.len()
+            ))),
+        }
+    }
+
+    fn evaluate(&self, inputs: &[Series]) -> DaftResult<Series> {
+        let result_name = inputs[0].name();
+        match (inputs[0].data_type(), inputs[1].data_type()) {
+            (DataType::Binary, DataType::Binary) => {
+                let left_array = inputs[0].downcast::<BinaryArray>()?;
+                let right_array = inputs[1].downcast::<BinaryArray>()?;
+                let result = left_array.binary_concat(right_array)?;
+                Ok(
+                    BinaryArray::from((result_name, Box::new(result.as_arrow().clone())))
+                        .into_series(),
+                )
+            }
+            (DataType::Binary, DataType::Null) | (DataType::Null, DataType::Binary) => {
+                // If either input is null, return a null array of the same length
+                let len = inputs[0].len();
+                Ok(Series::full_null(result_name, &DataType::Binary, len))
+            }
+            (DataType::FixedSizeBinary(_), DataType::FixedSizeBinary(_)) => {
+                let left_array = inputs[0].downcast::<FixedSizeBinaryArray>()?;
+                let right_array = inputs[1].downcast::<FixedSizeBinaryArray>()?;
+                let result = left_array.binary_concat(right_array)?;
+                Ok(
+                    FixedSizeBinaryArray::from((result_name, Box::new(result.as_arrow().clone())))
+                        .into_series(),
+                )
+            }
+            (DataType::FixedSizeBinary(_), DataType::Binary)
+            | (DataType::Binary, DataType::FixedSizeBinary(_)) => {
+                let left_array = match inputs[0].data_type() {
+                    DataType::FixedSizeBinary(_) => inputs[0]
+                        .downcast::<FixedSizeBinaryArray>()?
+                        .cast(&DataType::Binary)?,
+                    _ => inputs[0].downcast::<BinaryArray>()?.clone().into_series(),
+                };
+                let right_array = match inputs[1].data_type() {
+                    DataType::FixedSizeBinary(_) => inputs[1]
+                        .downcast::<FixedSizeBinaryArray>()?
+                        .cast(&DataType::Binary)?,
+                    _ => inputs[1].downcast::<BinaryArray>()?.clone().into_series(),
+                };
+                let result = left_array.binary()?.binary_concat(right_array.binary()?)?;
+                Ok(
+                    BinaryArray::from((result_name, Box::new(result.as_arrow().clone())))
+                        .into_series(),
+                )
+            }
+            (DataType::FixedSizeBinary(_), DataType::Null)
+            | (DataType::Null, DataType::FixedSizeBinary(_)) => {
```

Comment:
Minor nit: you can remove a match branch by matching instead on `(_, DataType::Null) | (DataType::Null, _)` to cover both Binary and FixedSizeBinary cases